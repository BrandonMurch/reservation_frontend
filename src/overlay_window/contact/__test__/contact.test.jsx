// Dependencies
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import { create } from 'react-test-renderer';

// Components
import ContactForm from '../index';

describe('<ContactForm />', () => {
  let component;
  let mockSubmitFunction;
  let container = null;
  const reservation = {
    time: '21:00',
    date: '2020-10-09',
    partySize: 2,
  };

  beforeEach(async () => {
    container = document.createElement('div');
    document.body.appendChild(container);
    mockSubmitFunction = jest.fn();
    component = await render(
      <ContactForm onSubmit={mockSubmitFunction} reservation={reservation} />,
    );
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(
      <ContactForm onSubmit={mockSubmitFunction} reservation={reservation} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should display new user on page load', () => {
    const firstName = component.getByLabelText(/First Name/i);
    expect(firstName).toBeInTheDocument();
    const lastName = component.getByLabelText(/Last Name/i);
    expect(lastName).toBeInTheDocument();
    const email = component.getByLabelText(/Email/i);
    expect(email).toBeInTheDocument();
    const phone = component.getByLabelText(/Phone Number/i);
    expect(phone).toBeInTheDocument();
  });
  it('should display switch to login on button click', () => {
    const button = component.getByRole('button', { name: 'Login' });
    fireEvent.click(button);
    const email = component.getByLabelText(/Email/i);
    expect(email).toBeInTheDocument();
    const password = component.getByLabelText(/Password/i);
    expect(password).toBeInTheDocument();
  });
  // FIXME - need to fill form in test.
  // it('should display call onSubmit when submitted', () => {
  //   const button = component.getByRole('button', { name: 'Submit' });
  //   fireEvent.click(button);
  //   expect(mockSubmitFunction).toHaveBeenCalled();
  // });
});
