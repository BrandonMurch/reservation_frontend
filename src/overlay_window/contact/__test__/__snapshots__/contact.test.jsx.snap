// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> should match snapshot 1`] = `
<div
  className="container"
>
  <p
    className="title"
  >
    2 people
    <br />
    21:00
    <br />
    Fri Oct 09 2020
  </p>
  <p
    className="switchText"
  >
    Do you already have an account?
    <button
      className="switchButton"
      onClick={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      Login
    </button>
  </p>
  <form
    className="container"
    data-testid="form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="firstName"
      >
        First Name
      </label>
      <input
        className="input"
        id="firstName"
        name="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        pattern={null}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="lastName"
      >
        Last Name
      </label>
      <input
        className="input"
        id="lastName"
        name="lastName"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        pattern={null}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="username"
      >
        Email
      </label>
      <input
        className="input"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        pattern="^[\\\\w\\\\-_.+]*[\\\\w\\\\-_.]@([\\\\w]+\\\\.)+[\\\\w]+[\\\\w]$"
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="phoneNumber"
      >
        Phone Number
      </label>
      <input
        className="input"
        id="phoneNumber"
        name="phoneNumber"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        pattern="^\\\\+\\\\d{1,3} \\\\d{6,14}$"
        placeholder="Phone Number"
        required={true}
        type="tel"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="hiddenLabelText"
        htmlFor="comments"
      >
        Comments / Dietary requirements
      </label>
      <input
        className="input"
        id="comments"
        name="comments"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        pattern={null}
        placeholder="Comments / Dietary requirements"
        required={false}
        type="textarea"
        value=""
      />
    </div>
    <div
      className="inputGroup"
    >
      <label
        className="labelText"
        htmlFor="tAC"
      >
        Yes, I agree to the terms and conditions
      </label>
      <input
        className="input"
        id="tAC"
        name="tAC"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        pattern={null}
        placeholder="Yes, I agree to the terms and conditions"
        required={true}
        type="checkbox"
        value=""
      />
    </div>
    <input
      className="button"
      disabled={false}
      type="submit"
      value="Next"
    />
  </form>
</div>
`;
